tests/expr_and_series/cast_test.py

```diff
@@ -260,14 +262,14 @@ def test_cast_struct(request: pytest.FixtureRequest, constructor: Constructor) -
         native_df = native_df.withColumn(  # type: ignore[union-attr]
             "a",
             F.struct(
-                F.col("a.movie ").alias("movie ").cast(T.StringType()),
-                F.col("a.rating").alias("rating").cast(T.DoubleType()),
+                F.col("a.movie ").cast(T.StringType()).alias("movie "),
+                F.col("a.rating").cast(T.DoubleType()).alias("rating"),
             ),
         )
 
-    dtype = nw.Struct([nw.Field("movie ", nw.String()), nw.Field("rating", nw.Float64())])
-    result = nw.from_native(native_df).select(nw.col("a").cast(dtype)).lazy().collect()
+        assert nw.from_native(native_df).schema == nw.Schema({"a": dtype})
 
+    result = nw.from_native(native_df).select(nw.col("a").cast(dtype)).lazy().collect()
```

Comment:
I think it could make sense to cast to another dtype instead of the original one, e.g.:

`nw.Struct([nw.Field("movie ", nw.String()), nw.Field("rating", nw.Float32())])`

just to make sure that something happens 😁